### PR TITLE
[HangoutsExtension] Remove hangouts extension

### PR DESCRIPTION
**Pull request by jagadeshjai**
- Disable hangouts extension by passing the buildflag via GN args.
- Remove the option available for the same in setting page.

Resolves https://github.com/brave/brave-browser/issues/39660

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### Upgrade scenario
Snippet courtesy of https://fedi.simonwillison.net/@simon/112757810519145581

1. Have a version of Brave BEFORE this change - clean profile.
2. Visit brave://settings/extensions
3. `Hangouts` should be enabled by default. Toggle it to OFF. Then back to ON.
4. Upgrade to a version of Brave which has this fix
5. Verify `Hangouts` is no longer in brave://settings/extensions
6. Visit brave://extensions-internals/
7. Verify `nkeimhogjdpnpccoofpliimaahmaaome` is no longer there
8. Visit https://google.com
9. Open the dev tools
10. In the console, paste in the following and hit <kbd>Enter</kbd>:
    ```
    chrome.runtime.sendMessage(
      "nkeimhogjdpnpccoofpliimaahmaaome",
      { method: "cpu.getInfo" },
      (response) => {
        console.log(JSON.stringify(response, null, 2));
      },
    );
    ```
11. There should be an error shown in console and no information should show.

### Fresh profile
Same as `Upgrade scenario` - but skip to step 5

### CODE REVIEW

File: browser/brave_content_browser_client_browsertest.cc [hunk 1]
```diff
@@ -625,29 +626,6 @@ IN_PROC_BROWSER_TEST_F(BraveContentBrowserClientTest, MixedContentForOnion) {
 }
 #endif
 
-#if BUILDFLAG(ENABLE_HANGOUT_SERVICES_EXTENSION)
-IN_PROC_BROWSER_TEST_F(BraveContentBrowserClientTest,
-                       HangoutsDisabledByDefault) {
-  ASSERT_FALSE(browser()->profile()->GetPrefs()->GetBoolean(kHangoutsEnabled));
-  extensions::ExtensionRegistry* registry =
-      extensions::ExtensionRegistry::Get(browser()->profile());
-  ASSERT_FALSE(registry->enabled_extensions().Contains(hangouts_extension_id));
```
Comment (goodov) [inline browser/brave_content_browser_client_browsertest.cc:634]: how about converting this test into a new test with the "expected" list of extensions?

**Review by bsclifton — APPROVED** [review]
Changes LGTM; verified it works using some of the steps from test plan in https://github.com/brave/brave-core/pull/24583

String reviewers changes LGTM ++

CI run happening now with https://github.com/brave/brave-core/pull/24599

File: browser/brave_content_browser_client_browsertest.cc (hunk 1 above)
Comment (jagadeshjai) [reply browser/brave_content_browser_client_browsertest.cc:634]: Yes, I'll do that @goodov

**Comment by jagadeshjai** [comment]
@rebron  Could you please confirm whether we need to show a InfoBar message or similar to convey the users for removing this option.

File: browser/brave_content_browser_client_browsertest.cc (hunk 1 above)
Comment (jagadeshjai) [reply browser/brave_content_browser_client_browsertest.cc:634]: Added :+1:

**Comment by bsclifton** [comment]
@jagadeshjai I checked with @rebron - no infobar needed! 😄

**Comment by bsclifton** [comment]
Everything here looks good to go! 

@goodov it's ready for your re-review after @jagadeshjai added the test case 😄👍

File: components/constants/pref_names.h [hunk 2]
```diff
@@ -61,6 +61,7 @@ inline constexpr char kShieldsStatsBadgeVisible[] =
 inline constexpr char kAdControlType[] = "brave.ad_default";
 inline constexpr char kGoogleLoginControlType[] = "brave.google_login_default";
 inline constexpr char kWebTorrentEnabled[] = "brave.webtorrent_enabled";
+// Deprecated
```
Comment (goodov) [inline components/constants/pref_names.h:64]: I don't think this comment is needed here. right now it looks like everything below is deprecated.

File: browser/brave_content_browser_client_browsertest.cc [hunk 3]
```diff
@@ -730,3 +708,21 @@ IN_PROC_BROWSER_TEST_F(BraveContentBrowserClientReferrerTest,
...
+      extensions::kWebStoreAppId,
+      extension_misc::kPdfExtensionId,
+  };
+
+  extensions::ExtensionRegistry* registry =
+      extensions::ExtensionRegistry::Get(browser()->profile());
+  std::set<std::string> installed_extensions =
+      registry->GenerateInstalledExtensionsSet().GetIDs();
+
+  EXPECT_EQ(expected_extensions, installed_extensions);
```
Comment (goodov) [inline browser/brave_content_browser_client_browsertest.cc:727]: I wonder will this actually output the content of both sets if they're not the same?

I think this should output the actual content: `EXPECT_THAT(installed_extensions, testing::ElementsAreArray(expected_extensions));`. Can you try that?

**Review by goodov — APPROVED** [review]
lgtm with few nits

File: browser/brave_content_browser_client_browsertest.cc (hunk 3 above)
Comment (jagadeshjai) [reply browser/brave_content_browser_client_browsertest.cc:727]: Actually EXPECT_EQ will give the output of both Set if its not equal, I checked that before itself.

File: components/constants/pref_names.h (hunk 2 above)
Comment (jagadeshjai) [reply components/constants/pref_names.h:64]: @goodov How about adding right side of the definition?

```suggestion
inline constexpr char kWebTorrentEnabled[] = "brave.webtorrent_enabled"; // Deprecated
```

Comment (bsclifton) [reply components/constants/pref_names.h:64]: Webtorrent is OK; it's the one below that is deprecated

Comment (jagadeshjai) [reply components/constants/pref_names.h:64]: Yes yes, I just wanted to show it with example :laughing:

**Pull request merged**